Sources/CoreMetrics/Metrics.swift

```diff
@@ -279,7 +289,12 @@ public class Timer {
     ///     - value: Duration to record.
     @inlinable
     public func recordSeconds<DataType: BinaryInteger>(_ duration: DataType) {
-        self.recordNanoseconds(Int64(duration * 1_000_000_000))
+        let result = Int64(duration).multipliedReportingOverflow(by: 1_000_000_000)
+        if result.overflow {
+            self.recordNanoseconds(Int64.max)
+        } else {
+            self.recordNanoseconds(result.partialValue)
+        }
```

Comment:
capping these to Int64.max is a bit interesting, but i don't see much of an option unless we change the underlying storage type to Double

Reply:
`Int64` will hold almost 300 years worth of nanoseconds. I think that's good enough :)

Reply:
2^63/(1_000_000_000*3600*24*365) = 292.4....

Reply:
Yeah, the only case where hitting max happens in such counters is when many many simultanious updates of large values happen, for most metrics its a non problem I think.

Reply:
if we believe we need bigger numbers, maybe micro-second resolution is good enough? Then we'd be up to 292,000 years :)

Reply:
y i think we are fine, just wanted to highlight it that its strange in the sense that it can cap the numbers dramatically

Reply:
also, do you feel we should print a warning in these cases?

Reply:
I don’t think we have to do anything more in the API, an impl could decide to do “the right thing” if it detects the cap is reached. Metrics backends often have a concept of events as well, so they could emit that for example. It also needs state to avoid logging the warning every time the saturated metric is being hit. Best to leave it to impls i think